Framework/Core/include/Framework/InputSpec.h

```diff
@@ -20,19 +21,11 @@ namespace framework {
 /// input or in output. This can be used, for example to match
 /// specific payloads in a timeframe.
 struct InputSpec {
-  /// The lifetime of objects being referred here.
-  enum Lifetime {
-    Timeframe,
-    Condition,
-    QA,
-    Transient
-  };
-
   std::string binding;
   header::DataOrigin origin;
   header::DataDescription description;
-  header::DataHeader::SubSpecificationType subSpec;
-  enum Lifetime lifetime;
+  header::DataHeader::SubSpecificationType subSpec = 0;
+  enum Lifetime lifetime = Lifetime::Timeframe;
```

Comment:
Is it appropriate to default to `Lifetime::Timeframe`, or should `Lifetime::Unknown` be foreseen?